File: wcpelib-bukkit/src/main/java/top/wcpe/wcpelib/bukkit/command/entity/CommandArgument.java
package top.wcpe.wcpelib.bukkit.command.entity;

import lombok.Getter;

/**
 * 子命令参数实体
 *
 * @author WCPE
 * @date 2021年4月24日 下午3:36:31
 */
public class CommandArgument {
    /**
     * 参数名称
     */
    @Getter
    private final String name;
    /**
     * 参数介绍
     */
    @Getter
    private final String describe;
    /**
     * 参数忽略默认值
     */
    @Getter
    private final String ignoreArg;

    private CommandArgument(Builder builder) {
        this.name = builder.name;
        this.describe = builder.describe;
        this.ignoreArg = builder.ignoreArg;
    }

    public static class Builder {
        private final String name;
        private String describe;
        private String ignoreArg;

        public Builder(String name) {
            this.name = name;
        }

        public Builder describe(String describe) {
            this.describe = describe;
            return this;
        }

        public Builder ignoreArg(String ignoreArg) {
            this.ignoreArg = ignoreArg;
            return this;
        }

        public CommandArgument build() {
            return new CommandArgument(this);
        }
    }
}

File: wcpelib-bungeecord/src/main/kotlin/top/wcpe/wcpelib/bungeecord/command/CommandArgument.kt
package top.wcpe.wcpelib.bungeecord.command

import top.wcpe.wcpelib.common.command.CommandArgument

/**
 * 由 WCPE 在 2022/9/27 22:31 创建
 *
 * Created by WCPE on 2022/9/27 22:31
 *
 * GitHub  : https://github.com/wcpe
 * QQ      : 1837019522
 * @author : WCPE
 * @since  : v1.1.0-alpha-dev-4
 */
@Deprecated("replace common command v2")
class CommandArgument private constructor(
    override val name: String,
    override val describe: String,
    override val ignoreArg: String?
) : CommandArgument {

    class Builder(private val name: String) {
        private var describe: String = ""
        private var ignoreArg: String? = null

        fun describe(describe: String): Builder {
            this.describe = describe
            return this
        }

        fun ignoreArg(ignoreArg: String): Builder {
            this.ignoreArg = ignoreArg
            return this
        }

        fun build(): CommandArgument {
            return CommandArgument(name, describe, ignoreArg)
        }
    }
}